ocrd_utils/ocrd_utils/logging.py

```diff
@@ -91,7 +92,13 @@ def getLogger(*args, **kwargs):
     Wrapper around ``logging.getLogger`` that respects `overrideLogLevel <#setOverrideLogLevel>`_.
     """
     if not _initialized_flag:
-        raise Exception("No initLogging() before getLogger()")
+        # XXX logger.exception may only be called in an except clause
+        initLogging()
+        logging.getLogger('').critical('*** getLogger was called before initLogging ***')
+        logging.getLogger('').critical('*** If you encounter this when running a processor, open an issue about it in the resp. project on GitHub ***')
+        for x in format_stack(limit=2):
+            for y in x.split('\n'):
+                logging.getLogger('').critical(y)
```

Comment:
Isn't it better to just
```suggestion
        logging.getLogger('').critical('getLogger must not be called before initLogging', stack_info=True)
```

Reply:
Wasn't aware of that, thanks. I want to convey 2 bits of information though: That there is an issue with a processor and where the call happens, so it can be fixed in the processor. If you think that is implied by your asterisk-free less dramatic message, fine with me :)

Reply:
Unfortunately, `stack_info` will print a full stacktrace outside the logging lines. There is a `stacklevel` argument that would remedy this, but it's only available since 3.8.

Reply:
> If you think that is implied by your asterisk-free less dramatic message, fine with me :)

I am fine with your text, I just wanted to avoid the `format_stack` stuff. But as you found out...

> Unfortunately, `stack_info` will print a full stacktrace outside the logging lines. There is a `stacklevel` argument that would remedy this, but it's only available since 3.8.

Yes, then this is not really an option.

Reply:
I've simplified the stacktrace wrangling to 2 lines though.